### PR TITLE
Allow use gluster recipe on already existed filesystem

### PR DESCRIPTION
**Pull request by theundefined**

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license          'Apache 2.0'
 description      'Installs and configures Gluster servers and clients'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version          '5.1.0'
+version          '5.1.1'
```
Comment (shortdudey123) [inline metadata.rb]: Revert

File: recipes/server_setup.rb [hunk 2]
```diff
@@ -32,21 +32,32 @@
...
-      else
-        Chef::Log.warn('No filesystem specified, defaulting to xfs')
-        filesystem = 'xfs'
-      end
-      # Even though this says volume_name, it's actually Brick Name. At the moment this method only supports one brick per volume per server
-      logical_volume volume_name do
-        size volume_values['size']
-        filesystem filesystem
-        mount_point "#{node['gluster']['server']['brick_mount_path']}/#{volume_name}"
+    if node['gluster']['server']['disks'].any?
```
Comment (shortdudey123) [inline recipes/server_setup.rb]: Can you add an `else` to this with a warning log saying that no LVM setup will be done?

File: recipes/server_setup.rb [hunk 3]
```diff
@@ -32,21 +32,32 @@
...
+        # Even though this says volume_name, it's actually Brick Name. At the moment this method only supports one brick per volume per server
+        logical_volume volume_name do
+          size volume_values['size']
+          filesystem filesystem
+          mount_point "#{node['gluster']['server']['brick_mount_path']}/#{volume_name}"
+        end
       end
     end
+
+    directory "#{node['gluster']['server']['brick_mount_path']}/#{volume_name}" do
```
Comment (shortdudey123) [inline recipes/server_setup.rb]: Have you confirmed this will not have any affect on LVM based mounts?

File: recipes/server_setup.rb [hunk 4]
```diff
@@ -92,6 +103,7 @@
       # Create option string
       force = false
       options = ''
+      force = true if node['gluster']['server']['disks'].empty?
```
Comment (shortdudey123) [inline recipes/server_setup.rb]: Why is force need here? What error does gluster through?

Comment (theundefined) [inline recipes/server_setup.rb]: force is needed only when gluster is creating on root partition/volume (for example on kitchen ;). But i dont know how to easy check if there is that case. I think about conditional "force" option for volume configuration.

File: recipes/server_setup.rb (hunk 3 above)
Comment (theundefined) [inline recipes/server_setup.rb]: i'ts just creating directory. If directory already exists - nothing will change. But - i can put it into conditional block.

File: metadata.rb (hunk 1 above)
Comment (theundefined) [inline metadata.rb]: why? I think that's a "feature" fix, so it's easier to bump version to point which version works :)

File: recipes/server_setup.rb (hunk 2 above)
Comment (theundefined) [inline recipes/server_setup.rb]: ok

File: recipes/server_setup.rb (hunk 4 above)
Comment (shortdudey123) [inline recipes/server_setup.rb]: `df /path/to/dir | tail -1` then check the last part of the array
There is probably other solutions too

Example outside the context of chef
```
[3] pry(main)> `df /etc | tail -1`.chomp.split[-1]
=> "/"
[4] pry(main)> 
```

File: recipes/server_setup.rb (hunk 3 above)
Comment (shortdudey123) [inline recipes/server_setup.rb]: Cool, just confirming

File: metadata.rb (hunk 1 above)
Comment (theundefined) [inline metadata.rb]: ok

**Comment by theundefined** [comment]
Is that better? :-)

**Comment by theundefined** [comment]
squashed :-)

**Pull request merged**